### PR TITLE
[BUGFIX] Make template file resolving recursive

### PR DESCRIPTION
**Pull request by NamelessCoder**
Fixes resolving of files in subfolders, among other
things making all resolveAvailableXyz methods
return all files in subfolders.

### CODE REVIEW

**Comment by NamelessCoder** [comment]
Don't merge - will amend with recursion limit to prevent infinite loops in case someone made some unfortunate symlinks.

**Comment by NamelessCoder** [comment]
Updated to use a filtered recursive directory iterator. This iterator has built-in infinite symlink protection - tested and confirmed.

**Comment by opi99** [comment]
Code change looks good from my site, any way to test this change inside the cache warm up?

**Comment by NamelessCoder** [comment]
You can test this with cache warmup, if you have a partial template in a subfolder then without this change, you'd see no compiled template for it - with it, you'd see a compiled class named like the partial (with an underscore to replace the slash).

**Comment by NamelessCoder** [comment]
@mbrodala If this still builds green and you find the patch okay would you mind merging this one?

**Comment by mbrodala** [comment]
I'll try to add a unit test first.

**Comment by mbrodala** [comment]
@NamelessCoder Can you first rebase onto the latest master?

File: src/View/TemplatePaths.php [hunk 1]
```diff
@@ -325,8 +325,13 @@ protected function resolveFilesInFolders(array $folders, $format)
      */
     protected function resolveFilesInFolder($folder, $format)
     {
-        $files = glob($folder . '*.' . $format);
-        return !$files ? [] : $files;
+        if (!is_dir($folder)) {
+            return [];
+        }
+        $directoryIterator = new \RecursiveDirectoryIterator($folder, \FilesystemIterator::FOLLOW_SYMLINKS | \FilesystemIterator::SKIP_DOTS);
```
Comment (wmdbMattes) [inline src/View/TemplatePaths.php]: Would it be viable to run Symfony Finder here?

Comment (NamelessCoder) [reply src/View/TemplatePaths.php]: Except for Fluid not having any dependencies at all right now, no. I would be happy to reconsider this if we begin adding dependencies, but I'd not do that until we have to. IMHO it doesn't save enough code in this case to justify.

Comment (wmdbMattes) [reply src/View/TemplatePaths.php]: jup, that's reasonable

**Comment by NamelessCoder** [comment]
@mbrodala While a unit test surely would be appreciated, I don't consider it a problem if a patch that adds the test comes later. By now we have a few uncovered places so a general update to tests is required at some point anyway. We've got tests aplenty to ensure that non-recursive resolving works ;)

**Comment by mbrodala** [comment]
The test is green now, please have another look. If you're fine, I'll merge this.

**Comment by NamelessCoder** [comment]
Test looks good to me - would work the same with any of the three resource types. Thanks for taking care Mathias!

**Pull request merged**